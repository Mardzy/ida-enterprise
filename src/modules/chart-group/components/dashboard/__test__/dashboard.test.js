import { fromJS } from "immutable";
import Dashboard, { CHART_GROUP_SECTIONS } from "../dashboard";

describe("The chart group dashboard", () => {

  it("renders without passing any props", () => {
    const wrapper = mount(<Dashboard/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("uses a custom chart renderer function", () => {
    const renderChart = chart => chart.get("config");

    const sections = fromJS({
      [CHART_GROUP_SECTIONS.main]: [{ config: "abcd" }],
      [CHART_GROUP_SECTIONS.left]: [{ config: "asdf" }],
      [CHART_GROUP_SECTIONS.right]: [{ config: "sdfg" }]
    });

    const wrapper = mount(<Dashboard renderChart={renderChart} sections={sections}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});