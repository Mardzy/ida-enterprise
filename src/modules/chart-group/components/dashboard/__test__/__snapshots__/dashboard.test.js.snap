// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The chart group dashboard renders without passing any props 1`] = `
<Dashboard
  fetchData={[Function]}
  renderChart={[Function]}
  sections={
    Immutable.Map {
      "main": Immutable.List [],
      "left": Immutable.List [],
      "right": Immutable.List [],
    }
  }
>
  <MainWithSecondary
    main={[Function]}
    secondary={[Function]}
  >
    <SingleColumn
      content={[Function]}
    >
      <Grid
        columns={1}
        stackable={true}
      >
        <div
          className="ui stackable one column grid"
        >
          <GridColumn>
            <div
              className="column"
            >
              <Component />
            </div>
          </GridColumn>
        </div>
      </Grid>
    </SingleColumn>
    <DoubleColumn
      left={[Function]}
      right={[Function]}
    >
      <Grid
        columns={2}
        stackable={true}
      >
        <div
          className="ui stackable two column grid"
        >
          <GridColumn
            computer={8}
            tablet={16}
          >
            <div
              className="eight wide computer sixteen wide tablet column"
            >
              <Component />
            </div>
          </GridColumn>
          <GridColumn
            computer={8}
            tablet={16}
          >
            <div
              className="eight wide computer sixteen wide tablet column"
            >
              <Component />
            </div>
          </GridColumn>
        </div>
      </Grid>
    </DoubleColumn>
  </MainWithSecondary>
</Dashboard>
`;

exports[`The chart group dashboard uses a custom chart renderer function 1`] = `
<Dashboard
  fetchData={[Function]}
  renderChart={[Function]}
  sections={
    Immutable.Map {
      "main": Immutable.List [
        Immutable.Map {
          "config": "abcd",
        },
      ],
      "left": Immutable.List [
        Immutable.Map {
          "config": "asdf",
        },
      ],
      "right": Immutable.List [
        Immutable.Map {
          "config": "sdfg",
        },
      ],
    }
  }
>
  <MainWithSecondary
    main={[Function]}
    secondary={[Function]}
  >
    <SingleColumn
      content={[Function]}
    >
      <Grid
        columns={1}
        stackable={true}
      >
        <div
          className="ui stackable one column grid"
        >
          <GridColumn>
            <div
              className="column"
            >
              abcd
            </div>
          </GridColumn>
        </div>
      </Grid>
    </SingleColumn>
    <DoubleColumn
      left={[Function]}
      right={[Function]}
    >
      <Grid
        columns={2}
        stackable={true}
      >
        <div
          className="ui stackable two column grid"
        >
          <GridColumn
            computer={8}
            tablet={16}
          >
            <div
              className="eight wide computer sixteen wide tablet column"
            >
              asdf
            </div>
          </GridColumn>
          <GridColumn
            computer={8}
            tablet={16}
          >
            <div
              className="eight wide computer sixteen wide tablet column"
            >
              sdfg
            </div>
          </GridColumn>
        </div>
      </Grid>
    </DoubleColumn>
  </MainWithSecondary>
</Dashboard>
`;
